### PR TITLE
:lipstick: added routing and made header components

### PR DESCRIPTION
**Pull request by ivanms1**
Don't forget to create the pull request @revengemiroz :wink:

### CODE REVIEW

File: client/src/Context/navColor.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import React, { createContext, useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+
+export const NavColorContext = createContext()
+
+function navColor({ children }) {
+    const [isSelected, setIsSelected] = useState(false)
+
+    useEffect(() => {
+        const params = useParams();
```
Comment (ivanms1) [inline client/src/Context/navColor.js]: any reason to use it inside a useEffect?

File: client/src/Context/navColor.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    }, [])
+
+    function colorChange(params) {
+        switch (params) {
+            case 'contact':
```
Comment (ivanms1) [inline client/src/Context/navColor.js]: are you planning to add more cases here?

File: client/src/Routes/Routes.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import Submitproject from '../pages/SubmitYourProject/Submit';
+import Weekly from '../pages/WeeklyProjects/Weekly';
+import Error from '../pages/Error/error';
+
+import { Switch, Route } from 'react-router-dom';
+
+function Routes(props) {
+    return (
+
+
```
Comment (ivanms1) [inline client/src/Routes/Routes.js]: empty space

File: client/src/assets/kathmandu.svg [hunk 4]
```diff
@@ -0,0 +1,116 @@
+<svg width="60" height="60" viewBox="0 0 60 60" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
```
Comment (ivanms1) [inline client/src/assets/kathmandu.svg]: can you svgo this svg? https://jakearchibald.github.io/svgomg/
for optimization

File: client/src/components/Header/Header.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
+import React, { forwardRef } from 'react';
+import { Container, Links, Logo, StyledLink } from "./style";
+import Kathmandu from '../../assets/kathmandu.svg';
+import MobileMenu from '../MobileMenu/Mobilemenu';
+import BurgerIcon from '../BurgerIcon/BurgerIcon';
+import Popup from 'reactjs-popup';
```
Comment (ivanms1) [inline client/src/components/Header/Header.js]: just a convention, imports from libraries should come before imports from the app.

File: client/src/components/MobileMenu/Mobilemenu.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { Menu } from './style';
+import { NavLink } from 'react-router-dom';
+
+export default function Mobilemenu({ close }) {
+    return (
+        <Menu>
+            <ul>
+                <li>
+                    <NavLink onClick={close} activeClassName='current' style={{ textDecoration: 'none', color: '#474547' }} to='/'>
```
Comment (ivanms1) [inline client/src/components/MobileMenu/Mobilemenu.js]: any reason for not using styled components here?
for consistency

File: client/src/components/Header/style.js [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+    font-weight:600;
+    letter-spacing:2px;
+}
+
+
+
+
+@media screen and (max-width:535px){
+   /* border:2px solid green; */
+display:flex;
```
Comment (ivanms1) [inline client/src/components/Header/style.js]: indentation

File: client/src/components/Header/style.js [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+   justify-content:center;
+   align-items:center;
+}
+
+
+`
+
+
+export const Logo = styled.div`
+/* border:2px solid green; */
```
Comment (ivanms1) [inline client/src/components/Header/style.js:35]: indentation

File: client/src/components/Header/style.js [hunk 9]
```diff
@@ -0,0 +1,127 @@
+import styled from 'styled-components';
+import { Link } from 'react-router-dom';
+
+export const Container = styled.div`
+border-bottom:1px solid #6C757D;
```
Comment (ivanms1) [inline client/src/components/Header/style.js:5]: indentation

File: client/src/components/BurgerIcon/style.js [hunk 10]
```diff
@@ -0,0 +1,40 @@
+import styled from 'styled-components';
+
+export const Container = styled.div`
+  @media (min-width: 595px) {
```
Comment (ivanms1) [inline client/src/components/BurgerIcon/style.js:4]: there is indentation on this first element for not in the ones below need to pick one

**Comment by ivanms1** [comment]
maybe check if we can add a linter for styles components and added to the workspace settings, good job so far.

File: client/src/Context/navColor.js (hunk 1 above)
Comment (revengemiroz) [reply client/src/Context/navColor.js]: it was just experimental code will scrap the context side for now.

File: client/src/Context/navColor.js (hunk 2 above)
Comment (revengemiroz) [reply client/src/Context/navColor.js]: this is the same as experimental will scrap it.

File: client/src/Routes/Routes.js (hunk 3 above)
Comment (revengemiroz) [reply client/src/Routes/Routes.js]: will clean up that

File: client/src/assets/kathmandu.svg (hunk 4 above)
Comment (revengemiroz) [reply client/src/assets/kathmandu.svg]: sure ill look into that

File: client/src/components/Header/Header.js (hunk 5 above)
Comment (revengemiroz) [reply client/src/components/Header/Header.js]: ah got it... will make adjustments

File: client/src/components/MobileMenu/Mobilemenu.js (hunk 6 above)
Comment (revengemiroz) [reply client/src/components/MobileMenu/Mobilemenu.js]: i can refactor it again but there was an issue where the style wouldnt wrap around the router doms Link and NavLink. Solved the issue in another routes. need to refactor here.

File: client/src/components/Header/style.js (hunk 7 above)
Comment (revengemiroz) [reply client/src/components/Header/style.js]: got it.

File: client/src/components/BurgerIcon/style.js (hunk 10 above)
Comment (revengemiroz) [reply client/src/components/BurgerIcon/style.js:4]: got it

File: client/src/components/MobileMenu/Mobilemenu.js (hunk 6 above)
Comment (revengemiroz) [reply client/src/components/MobileMenu/Mobilemenu.js]: :pencil: fixed the styling on this

File: client/src/Context/navColor.js (hunk 1 above)
Comment (revengemiroz) [reply client/src/Context/navColor.js]: :pencil:

File: client/src/Routes/Routes.js (hunk 3 above)
Comment (revengemiroz) [reply client/src/Routes/Routes.js]: :pencil:

File: client/src/assets/kathmandu.svg (hunk 4 above)
Comment (revengemiroz) [reply client/src/assets/kathmandu.svg]: :pencil:

File: client/src/components/Header/Header.js (hunk 5 above)
Comment (revengemiroz) [reply client/src/components/Header/Header.js]: :pencil:

File: client/src/components/Header/Header.js [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+import { useLocation } from 'react-router-dom';
+import { Container, Links, Logo, StyledLink } from "./style";
+import Kathmandu from '../../assets/kathmandu.svg';
+import MobileMenu from '../MobileMenu/Mobilemenu';
+import BurgerIcon from '../BurgerIcon/BurgerIcon';
+
+function Header(props) {
+
+    let location = useLocation()
+    console.log(location.pathname)
```
Comment (ivanms1) [inline client/src/components/Header/Header.js]: remove please

**Review by revengemiroz — APPROVED** [review]

**Review by revengemiroz — APPROVED** [review]

**Pull request merged**